### PR TITLE
[FIX] web: hide action button when no action

### PR DESCRIPTION
**Pull request by pka-odoo**
PURPOSE

when try to see time off analysis from employee, and click on "Action" button it
shows blank dropdown option.

SPECIFICATION

it was checking for the registry action data on the click on action button, so
sometime it was showing blank dropdown.
But now it will check the registry action if it have data or not before adding
it to the action dropdown

LINKS

Task:
https://www.odoo.com/web#id=2325587&action=4043&model=project.task&view_type=form&cids=2&menu_id=4720

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by Arcasias** [comment]
@aab-odoo could you review this? If you're ok with it I'll start working on a proper fix in master (see commit message)
@Polymorphe57 if you want to take a look

File: addons/google_drive/static/src/js/gdrive.js [hunk 1]
```diff
@@ -4,6 +4,28 @@ odoo.define('google_drive.ActionMenus', function (require) {
     const ActionMenus = require('web.ActionMenus');
     const DropdownMenuItem = require('web.DropdownMenuItem');
 
+    /**
+     * Fetches the google drive action menu item props.
+     * @param {Object} props
+     * @param {Object} env
+     * @param {Object} rpc
+     * @returns {Object | boolean} item props or false
```
Comment (aab-odoo) [inline addons/google_drive/static/src/js/gdrive.js:12]: document what is used in props and env

File: addons/web/static/src/js/components/action_menus.js [hunk 2]
```diff
@@ -21,6 +23,16 @@ odoo.define('web.ActionMenus', function (require) {
      */
     class ActionMenus extends Component {
 
+        async willStart() {
+            this.actionItems = await this._setActionItems(this.props);
+            this.printItems = this._setPrintItems(this.props);
```
Comment (aab-odoo) [inline addons/web/static/src/js/components/action_menus.js:28]: no action/print symmetry?

File: addons/web/static/src/js/components/action_menus.js [hunk 3]
```diff
@@ -30,59 +42,65 @@ odoo.define('web.ActionMenus', function (require) {
         }
 
         //---------------------------------------------------------------------
-        // Getters
+        // Private
         //---------------------------------------------------------------------
 
         /**
-         * @returns {Object[]}
+         * Add teh tooltips to the items
```
Comment (aab-odoo) [inline addons/web/static/src/js/components/action_menus.js:49]: the

File: addons/web/static/src/js/components/action_menus.js [hunk 4]
```diff
@@ -30,59 +42,65 @@ odoo.define('web.ActionMenus', function (require) {
...
-                ({ Component, getProps }, index) => ({
-                    key: `registry-action-${index}`,
-                    Component,
-                    props: getProps(this.props),
-                })
-            );
+            const registryActions = [];
+            const rpc = this.rpc.bind(this);
+            for (const { Component, getProps } of this.constructor.registry.values()) {
+                const itemProps = await getProps(props, this.env, rpc);
```
Comment (aab-odoo) [inline addons/web/static/src/js/components/action_menus.js:78]: falsy value means we should not render the item. I'm not found of this. However, I understand that we are constrained here as we don't want to adapt all getProps functions. In your version of this fix for master, we'll need to find a better API (if this is still relevant)

**Comment by aab-odoo** [comment]
robodoo r+

**Pull request closed without merging**